addon/mixins/map-route-metadata-ids-handler.js

```diff
@@ -0,0 +1,35 @@
+/**
+  @module ember-flexberry-gis-dummy
+*/
+
+import Ember from 'ember';
+
+/**
+  Mixin containing metadata ids as query params.
+
+  @class MapRouteMetadataIdsHandlerMixin
+*/
+export default Ember.Mixin.create({
```

Comment:
Mixin is unnecessary, it's logic can be included straight into `addon/routes/edit-map.js`.
Besides there is no logic to load metadata included into `metadata=1,2,3,4,5` URL-param.
You must override routes `model() {...}` hook, load all specified metadata, create new layers on matadata's basis, push them into model.mapLayer collection, and only after that resolve promise returned form `model() {...}` hook with loaded map model.